### DISCUSSION TITLE
Leveraging Zodios strong contract without a HTTP server

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#210] [Ideas]
Think about this scenario: you have an API exposed to the user, but also a BFF(Backend-for-Frontend), maybe using Next.js/Prisma/SvelteKit, where usually you'd be using DB access directly and writing custom queries, but you could be reusing Zodios strong contract just skipping the extra jump to an HTTP server.

**Comment by ecyrbe** [comment]
This is already existing in remix

**Reply by thelinuxlich** [reply]
This exists on any of these fullstack frameworks, the missing piece is Zodios server-side enforcing the API contract without a HTTP server

**Reply by ecyrbe** [reply]
Not what i ment. I you need B4F full typesafety, use remix. They have everything to enforce end to end typesafety, even simpler than zodios.

But, Zodios will stick to REST over HTTP use cases.

**Reply by thelinuxlich** [reply]
Yeah just thinking about the duplication
